js/views-manager.js

```diff
@@ -0,0 +1,99 @@
+// TODO:FIXME: 
+// search result layout
+// auto refresh?
+// issues view: coming soon
+
+var IssueBtnClickHandler = function(event) {
+  var issue = $(this).data("issue");
+  console.log("issue = ", issue);
+};
+
+var ViewsManager = {
+  projects: [],
+  $controlsContainer: $("#project-container .controls"),
+  $projectsContainer: $("#project-container .projects"),
+  renderProjectsIntoView: function(projects) {
+    this.$projectsContainer.hide();
+    $.each(projects, function(index, project) {
+      ProjectCard.render(project);
+    });
+    this.$projectsContainer.fadeIn();
+  },
+  renderAllProjectsIntoView: function() {
+    this.renderProjectsIntoView(this.projects);
+  },
+  showSingleProjectView: function(projectId) {
+    this.resetView();
+    this.renderAllProjectsIntoView();
+    $(".project").hide();
+    $(".project[data-id="+projectId+"]").addClass("single").show();
+    // FIXME: just a quick solution to add id query param to URL
+    window.history.pushState(projectId, "", window.location.href.split("?")[0] + "?id=" + projectId);
```

Comment:
not sure if i understand ^ correctly @cadecairos but i think this `pushState` line can be removed.